types/underscore/index.d.ts

```diff
@@ -5938,78 +5938,78 @@ declare module _ {
         * Wrapped type `any`.
         * @see _.identity
         **/
-        identity(): _Chain<T>;
+        identity(): _Chain<T, V>;
 
         /**
         * Wrapped type `any`.
         * @see _.constant
         **/
-        constant(): _Chain<T>;
+        constant(): _ChainSingle<() => V>;
 
         /**
         * Wrapped type `any`.
         * @see _.noop
         **/
-        noop(): _Chain<T>;
+        noop(): _ChainSingle<undefined>;
 
         /**
         * Wrapped type `number`.
         * @see _.times
         **/
-        times<TResult>(iterator: (n: number) => TResult, context?: any): _Chain<T>;
+        times<TResult>(iterator: (n: number) => TResult, context?: any): _Chain<TResult, TResult[]>;
 
         /**
         * Wrapped type `number`.
         * @see _.random
         **/
-        random(): _Chain<T>;
+        random(): _ChainSingle<number>;
         /**
         * Wrapped type `number`.
         * @see _.random
         **/
-        random(max: number): _Chain<T>;
+        random(max: number): _ChainSingle<number>;
 
         /**
         * Wrapped type `object`.
         * @see _.mixin
         **/
-        mixin(): _Chain<T>;
+        mixin(): _Chain<T, V>;
 
         /**
         * Wrapped type `string|Function|Object`.
         * @see _.iteratee
         **/
-        iteratee(context?: any): _Chain<T>;
+        iteratee(context?: any): _ChainSingle<any>;
 
         /**
         * Wrapped type `string`.
         * @see _.uniqueId
         **/
-        uniqueId(): _Chain<T>;
+        uniqueId(): _ChainSingle<string>;
 
         /**
         * Wrapped type `string`.
         * @see _.escape
         **/
-        escape(): _Chain<T>;
+        escape(): _ChainSingle<string>;
 
         /**
         * Wrapped type `string`.
         * @see _.unescape
         **/
-        unescape(): _Chain<T>;
+        unescape(): _ChainSingle<string>;
 
         /**
         * Wrapped type `object`.
         * @see _.result
         **/
-        result(property: string, defaultValue?: any): _Chain<T>;
+        result(property: string, defaultValue?: any): _ChainSingle<any>;
```

Comment:
Maybe possible to be more accurate:

```suggestion
        result<K in keyof V, D>(property: K, defaultValue?: D): _ChainSingle<
            V[K] extends () => infer R ? R | D : V[K] | D
        >;
        result<K not in keyof V, D = undefined>(property: K, defaultValue?: D): _ChainSingle<D>;
```

Reply:
Doing this would have to involve a bigger refactoring, since right now _ChainSingle is also broken in various ways.

Reply:
Let leave that one be for now